### PR TITLE
dns: z bit transactions and logging - v1

### PR DESCRIPTION
**Pull request by jasonish**
- Create DNS transaction even if z flag is set.
- Log z flag.

Ticket: 4515 4924

Pulls in PR: https://github.com/OISF/suricata/pull/6187

suricata-verify-pr:617

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 5389

File: doc/userguide/output/eve/eve-json-format.rst [hunk 1]
```diff
@@ -436,6 +436,7 @@ Outline of fields seen in the different kinds of DNS events:
 * "tc": Indicating in case of DNS answer flag, Truncation flag (ex: true if set)
 * "rd": Indicating in case of DNS answer flag, Recursion Desired flag (ex: true if set)
 * "ra": Indicating in case of DNS answer flag, Recursion Available flag (ex: true if set)
+* "z": Indicating in case of DNS answer flag, Reserved bit  (ex: true if set)
```
Comment (jlucovsky) [inline doc/userguide/output/eve/eve-json-format.rst:439]: nit: 2 spaces before the left paren.

Comment (victorjulien) [reply doc/userguide/output/eve/eve-json-format.rst:439]: fixed this up in my next branch

**Comment by victorjulien** [comment]
Merged in #6744, thanks!

**Pull request closed without merging**